src/library_pthread.js

```diff
@@ -36,13 +36,14 @@ var LibraryPThread = {
     initMainThreadBlock: function() {
       if (ENVIRONMENT_IS_PTHREAD) return undefined;
 
-#if PTHREAD_POOL_SIZE > 0
-      var requestedPoolSize = {{{ PTHREAD_POOL_SIZE }}};
+#if PTHREAD_POOL_SIZE
+      if (pthreadPoolSize > 0) {
 #if PTHREADS_DEBUG
-      out('Preallocating ' + requestedPoolSize + ' workers.');
-#endif
-      PThread.preallocatedWorkers = PThread.createNewWorkers(requestedPoolSize);
+        out('Preallocating ' + pthreadPoolSize + ' workers.');
 #endif
+        PThread.preallocatedWorkers = PThread.createNewWorkers(pthreadPoolSize);
+      }
+#endif // PTHREAD_POOL_SIZE
```

Comment:
I am not sure why this code block even exists.. this seems to double allocate a new pool of Workers that has already been created in the code below in `src/preamble.js`?